### DISCUSSION TITLE
Websocket connection blocked by firewall. Unable to edit documents

### DISCUSSION THREAD

**Discussion by entnx-01** [outline/outline#5750] [Self Hosting]
Dear community, 

I have a self hosted outline running for a while without any issues. From a remote location with an unknown firewall configuration it looks like all websocket connections are blocked. Without that connection I am unable to edit any page. Viewing a page works. 

Outline runs behind a nginx reverse proxy. 

As I can see from chromes dev console the connection is made with ssl (wss). 

Error Message:
`WebSocket connection to <my url> failed: Error during WebSocket handshake: Unexpected response code: 400`

There are also some errors with a different http code: 
`WebSocket connection to <my url> failed: Error during WebSocket handshake: Unexpected response code: 200`

Changing the firewall in any way is not an option, unfo. But I would have expected the websocket connection to have some kind of fallback to http polling?

**Comment by N4v41** [comment]
Hello @entnx-01, are you using https? usually when you are using https the conection is encrypted and there is no blocking.

i think that you can just disable the seemless editing feature on the options and it will work, if the websocket is the problem.

**Reply by entnx-01** [reply]
Hey @N4v41, 

thanks for your answer :) 

Yes I am using https. An I have also tried to disable seamless editing feature, but still the same error. 

As I have no access to the firewall and there will be no support from the firewall provider, its quiet hard to figure out the problem. 

Can someone confirm that outline should work without websocket connections? I guess I read somewhere its not working without websockets.

**Reply by N4v41** [reply]
@entnx-01 yeah, i was cheking, looks like since version [v0.67.1](https://github.com/outline/outline/releases/tag/v0.67.1) the editing is only over websockets.
